ontopy/excelparser.py

```diff
@@ -0,0 +1,267 @@
+
+                altlabels = row["altLabel"]
+                if isinstance(altlabels, str):
+                    altlabel_list = altlabels.split(";")
+                    for altlabel in altlabel_list:
+                        concept.altLabel.append(english(altlabel))
+
+                number_of_added_classes += 1
+
+            if number_of_added_classes == 0:
+                final_loop = True
+
+    # Add properties in a second loop
+    for _, row in data.iterrows():
+        properties = row["Relations"]
+        if isinstance(properties, str):
+            try:
+                concept = onto.get_by_label(row["prefLabel"])
+            except NoSuchLabelError:
+                pass
+            props = properties.split(";")
+            for prop in props:
+                try:
+                    concept.is_a.append(evaluate(onto, prop))
+                except pyparsing.ParseException as err:
+                    warnings.warn(
+                        f"Error in Property assignment for: {concept}. "
+                        f"Property to be Evaluated: {prop}. "
+                        f"Error is {err}."
+                    )
+
+    onto, catalog = get_metadata_from_dataframe(metadata, onto)
+    # Synchronise Python attributes to ontology
+    onto.sync_attributes(
+        name_policy="uuid", name_prefix="EMMO_", class_docstring="elucidation"
+    )
+    onto.dir_label = False
+
+    return onto, catalog
+
+
+# To test: with and without ontology as input
+def get_metadata_from_dataframe(  # pylint: disable=too-many-locals,too-many-branches,too-many-statements
+    metadata: pd.DataFrame,
+    onto: owlready2.Ontology = None,
+    base_iri_from_metadata: bool = True,
+    catalog: dict = None,
+) -> Tuple[owlready2.Ontology, dict]:
+    """
+    Populate ontology with metada from pd.DataFrame
+    """
+
+    if onto is None:
+        onto = get_ontology()
+
+    # base_iri from metadata if it exists and base_iri_from_metadata
+    if base_iri_from_metadata:
+        try:
+            base_iris = _parse_metadata_string(metadata, "Ontology IRI")
+            if len(base_iris) > 1:
+                warnings.warn(
+                    "More than one Ontology IRI given. " "The first was chosen."
+                )
+            base_iri = base_iris[0] + "#"
+            onto.base_iri = base_iri
+        except (TypeError, ValueError, AttributeError):
+            pass
+
+    # Get imported ontologies from metadata
+    try:
+        imported_ontology_paths = _parse_metadata_string(
+            metadata, "Imported ontologies"
+        )
+    except (TypeError, ValueError, AttributeError):
+        imported_ontology_paths = []
+    # Add imported ontologies
+    catalog = {} if catalog is None else catalog
+    for path in imported_ontology_paths:
+        imported = onto.world.get_ontology(path).load()
+        onto.imported_ontologies.append(imported)
+        catalog[imported.base_iri.rstrip("/")] = path
+
+    # Add title
```

Comment:
You can reduce the code duplication by defining a function as suggested above with `append_literal_cell()`